packages/test/test-utils/src/testFluidObjectInternal.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IFluidHandle,
+	IRequest,
+	IResponse,
+	type IFluidLoadable,
+} from "@fluidframework/core-interfaces";
+import { FluidObjectHandle } from "@fluidframework/datastore/internal";
+import {
+	IChannelFactory,
+	IFluidDataStoreRuntime,
+	type IChannel,
+} from "@fluidframework/datastore-definitions/internal";
+import {
+	IFluidDataStoreChannel,
+	IFluidDataStoreContext,
+} from "@fluidframework/runtime-definitions/internal";
+import { create404Response } from "@fluidframework/runtime-utils/internal";
+import type { ISharedObject } from "@fluidframework/shared-object-base/internal";
+
+/**
+ * A test Fluid object that will create a shared object for each key-value pair in the factoryEntries passed to load.
+ * The shared objects can be retrieved by passing the key of the entry to getSharedObject.
+ * It exposes the IFluidDataStoreContext and IFluidDataStoreRuntime.
+ * @remarks
+ * This is a simplified (does not use SharedMap) alternative to {@link TestFluidObject} which does not implement the external facing {@link ITestFluidObject} interface.
+ * @internal
+ */
+export class TestFluidObjectInternal implements IFluidLoadable {
+	public get IFluidLoadable() {
+		return this;
+	}
+
+	public readonly handle: IFluidHandle<this>;
+	private initializationPromise: Promise<void> | undefined;
+
+	/**
+	 * Creates a new TestFluidObjectInternal.
+	 * @param runtime - The data store runtime.
+	 * @param context - The data store context.
+	 * @param factoryEntries - A list of id to IChannelFactory mapping. For each item in the list,
+	 * a shared object is created which can be retrieved by calling getSharedObject() with the id;
+	 */
+	constructor(
+		public readonly runtime: IFluidDataStoreRuntime,
+		public readonly channel: IFluidDataStoreChannel,
+		public readonly context: IFluidDataStoreContext,
+		private readonly factoryEntriesMap: Map<string, IChannelFactory<ISharedObject>>,
```

Comment:
I would suggest calling this something that indicates these are the initial set of DDSes that will be created during the data store creation and initialization process only. Maybe `initialSharedObjects` or `rootSharedObjects`. The latter also indicates that these are rooted and will not be GC'ed.

The way I see usages is this tests would create a data store with an initial set of DDSes that are rooted. It may create additional DDSes after this point but those should not be rooted (not enforced today of course) and their handle should be stored in one of the root DDSes.
Unfortunately, I don't think there are documents that explain this usage pattern. We should consider adding them. The same pattern is recommended for data stores as well.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/34477